test/variables/static/multilocale.chpl

```diff
@@ -0,0 +1,14 @@
+proc add(arg) {
+  @functionStatic(sharingKind.computePerLocale)
+  ref counter = 0;
```

Comment:
It's not supposed to be legal to create a reference to a param. I'm guessing that this is an artifact of the static variable implementation, where we are heap-allocating something?

Can you use a ``ref`` return-intent instead? I think it would also be good to create an issue to capture this problem and also create a .future.